focus-android/app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt

```diff
@@ -612,7 +592,21 @@ class UrlInputFragment :
         BrowserSearch.searchCount["$defaultSearchEngineName.suggestion"].add()
     }
 
-    private fun openUrl(url: String, searchTerms: String?) {
+    private fun search(searchTerms: String) {
+        val tab = tab
+        if (tab != null) {
+            requireComponents.searchUseCases.defaultSearch.invoke(searchTerms, tab.id)
+            requireComponents.appStore.dispatch(AppAction.FinishEdit(tab.id))
+        } else {
+            requireComponents.searchUseCases.newPrivateTabSearch.invoke(
+                searchTerms = searchTerms,
+                private = true,
```

Comment:
Interesting, that's actual an API flaw here, that `newPrivateTabSearch` can be used for non-private searches. We should fix that, can you file a follow-up?